### ISSUE TITLE
Terragrunt "run-all plan" do not respects TERRAGRUNT_CONFIG env variable

### ISSUE THREAD

**Issue by Nek0trkstr** [gruntwork-io/terragrunt#2031]
Hi, I've set `TERRAGRUNT_CONFIG=.terragrunt.hcl`. 
So my root and leaf folders have this .terragrunt.hcl file.
Everything works well when I'm trying to run terragrunt from child folder, but I get "Could not find any subfolders with Terragrunt configuration files" error when trying to run from `terragrunt run-all plan` from root. Renaming .terragrunt.hcl -> terragrunt.hcl fixes the issue, but I would prefer to use "dot" prefix if possible.

It seems like [this line](https://github.com/gruntwork-io/terragrunt/blob/f6b5661906b71dfe2f261a029ae3809f233d06a7/config/config.go#L582) is using default config name and not the parsed argument.

**Comment by denis256** [comment]
Hi,
I AFAIK handling of hidden files between modules is not supported - hidden files are simply ignored during initialization of modules.

**Comment by Nek0trkstr** [comment]
Hi @denis256, thanks for response.
I think hidden files are still getting passed, because part of my root terragrunt config looks like this, and its working fine:
```
locals {
  # Automatically load account-level variables
  account_vars = read_terragrunt_config(find_in_parent_folders(".account.hcl"))

  # Automatically load region-level variables
  region_vars = read_terragrunt_config(find_in_parent_folders(".region.hcl"))

  # Extract the variables we need for easy access
  bucket    = local.account_vars.locals.bucket
  region   = local.region_vars.locals.region
}
```
Meaning I do load hidden files between modules.

**Comment by denis256** [comment]
Hello,
after a couple of tests, I think it is a bug since value of TERRAGRUNT_CONFIG is ignored

**Comment by denis256** [comment]
From what I see, the logic for identification of default terragrunt file should be updated to fetch value from `--terragrunt-config / TERRAGRUNT_CONFIG`

https://github.com/gruntwork-io/terragrunt/blob/master/config/config.go#L498

https://github.com/gruntwork-io/terragrunt/blob/master/config/config.go#L28

**Comment by Nek0trkstr** [comment]
Hi @denis256, thanks for testing this out!
I see you've labeled it as a "good first issue", I can try and fix that in next couple of days if it's ok.

**Comment by levkohimins** [comment]
Resolved in [v0.50.15](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.50.15) release.
